test/BaseTestEntityWithSources.cpp

```diff
@@ -82,4 +82,30 @@ void BaseTestEntityWithSources::testSourceVectorSetter() {
     std::vector<Source> deleter;
     da.sources(deleter);
     CPPUNIT_ASSERT(da.sourceCount() == 0);
-}
\ No newline at end of file
+}
+
+void BaseTestEntityWithSources::testSourceOrder() {
+    std::vector<std::string> names = { "source_a", "source_b", "source_c", "source_d", "source_e" };
```

Comment:
@gicmo is right, test will fail on random order. I  fixed the particular issue with the naming order but this does not generalize... :(
At this point we need a poll, methinks. What order should it be? Creation order? Name order? Order of things in the passed vector?

The easiest, and maybe best solution is to just take the order of the passed vector. This would mean that we simplify the whole function by removing all the old junk and simply setting the new stuff. We might even gain some performance in the hdf and fs backend (since we have quite a bit of file access there) but may loose some in a memory-backend...
 

If we want to have the creation order: is it the order in which we create the links to the entities or the time, the entities themselves have been created? Our created at will not be sufficient for such decisions and from the frontend we do not have access  to the real one...